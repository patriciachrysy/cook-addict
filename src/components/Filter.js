import React, { useState } from 'react';
import { useDispatch } from 'react-redux';
import { useNavigate } from 'react-router-dom';
import { getRecipesByFilter, startFetchingRecipe } from '../Redux/recipes/recipes';
import '../styles/Filter.css';

const Filter = () => {
  const dispatch = useDispatch();
  const navigate = useNavigate();
  const [tag, setTag] = useState('');
  const [keyword, setKeyword] = useState('');

  const handleSubmit = (e) => {
    e.preventDefault();
    document.body.click();
    if (tag.trim() || keyword.trim()) {
      dispatch(startFetchingRecipe());
      dispatch(getRecipesByFilter({ tag, keyword }));
      navigate('/recipes/search_results');
    }
  };

  return (
    <form onSubmit={(e) => handleSubmit(e)} className="filter-form">
      <div className="content">
        <input type="text" placeholder="Tag" value={tag} onChange={(e) => setTag(e.target.value)} />
        <input type="text" placeholder="Keyword" value={keyword} onChange={(e) => setKeyword(e.target.value)} />
        <button type="submit">Search</button>
      </div>
    </form>
  );
};

export default Filter;
