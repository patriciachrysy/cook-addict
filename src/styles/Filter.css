.filter-form .content {
  display: flex;
  flex-direction: column;
  align-items: flex-start;
}

.filter-form .content input {
  width: 87%;
  border: 1px solid black;
  background-color: #fff;
  border-radius: 4px;
  font-family: 'Rubik', sans-serif;
  color: black;
  letter-spacing: -0.15px;
  font-size: 1rem;
  padding: 0.7rem 1rem;
  margin: 5px 0;
}

.filter-form .content button {
  border: 2px solid black;
  font-weight: bold;
  align-self: flex-end;
  padding: 0.7rem 1rem;
  background-color: white;
  color: black;
  font-family: 'Rubik', sans-serif;
  font-size: 0.813rem;
  text-align: center;
  border-radius: 4px;
  letter-spacing: 0.5px;
  margin: 5px 0;
}
