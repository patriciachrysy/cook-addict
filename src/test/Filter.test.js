import React from 'react';
import renderer from 'react-test-renderer';

import { render, screen } from '@testing-library/react';
import { Provider } from 'react-redux';
import { BrowserRouter as Router } from 'react-router-dom';
import Filter from '../components/Filter';

import store from '../Redux/ConfigureStore';


it('renders correctly', () => {
  const tree = renderer.create(<Provider store={store}><Router><Filter /></Router></Provider>).toJSON();
  expect(tree).toMatchSnapshot();
});

test('Render Filter component', () => {
  render(<Provider store={store}><Router><Filter /></Router></Provider>);
  screen.debug();
});
