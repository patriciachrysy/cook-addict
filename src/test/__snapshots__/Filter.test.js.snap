// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<form
  className="filter-form"
  onSubmit={[Function]}
>
  <div
    className="content"
  >
    <input
      onChange={[Function]}
      placeholder="Tag"
      type="text"
      value=""
    />
    <input
      onChange={[Function]}
      placeholder="Keyword"
      type="text"
      value=""
    />
    <button
      type="submit"
    >
      Search
    </button>
  </div>
</form>
`;
